oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/OneandoneComputeServiceAdapter.java

```diff
@@ -62,77 +61,95 @@
+   private final String baremetalModelsKey = "BMC";
 
    @Inject
    OneandoneComputeServiceAdapter(OneAndOneApi api, CleanupResources cleanupResources,
+           GenerateHardwareRequest generateHardwareRequest,
            @Named(POLL_PREDICATE_SERVER) Predicate<Server> waitServerUntilAvailable,
            PasswordGenerator.Config passwordGenerator) {
       this.api = api;
       this.cleanupResources = cleanupResources;
+      this.generateHardwareRequest = generateHardwareRequest;
       this.waitServerUntilAvailable = waitServerUntilAvailable;
-      this.passwordGenerator = passwordGenerator;
    }
 
    @Override
    public NodeAndInitialCredentials<Server> createNodeWithGroupEncodedIntoName(String group, String name, Template template) {
 
-      final String dataCenterId = template.getLocation().getId();
+      final String dataCenterId ="908DC2072407C94C8054610AD5A53B8C";// template.getLocation().getId();
       Hardware hardware = template.getHardware();
       TemplateOptions options = template.getOptions();
       Server updateServer = null;
+      PasswordGenerator generate = new PasswordGenerator();
+      Types.ServerType serverType = Types.ServerType.cloud;
 
       final String loginUser = isNullOrEmpty(options.getLoginUser()) ? "root" : options.getLoginUser();
-      final String password = options.hasLoginPassword() ? options.getLoginPassword() : passwordGenerator.generate();
+      final String password = options.hasLoginPassword() ? options.getLoginPassword() : generate.generate();
       final String privateKey = options.hasLoginPrivateKey() ? options.getPrivateKey() : null;
       final org.jclouds.compute.domain.Image image = template.getImage();
       final int[] inboundPorts = template.getOptions().getInboundPorts();
-
-      //prepare hdds to provision
-      List<? extends Volume> volumes = hardware.getVolumes();
-      List<Hdd.CreateHdd> hdds = new ArrayList<Hdd.CreateHdd>();
-
-      for (final Volume volume : volumes) {
-         try {
-            //check if the bootable device has enough size to run the appliance(image).
-            float minHddSize = volume.getSize();
-            if (volume.isBootDevice()) {
-               SingleServerAppliance appliance = api.serverApplianceApi().get(image.getId());
-               if (appliance.minHddSize() > volume.getSize()) {
-                  minHddSize = appliance.minHddSize();
-               }
-            }
-            Hdd.CreateHdd hdd = Hdd.CreateHdd.create(minHddSize, volume.isBootDevice());
-            hdds.add(hdd);
-         } catch (Exception ex) {
-            throw Throwables.propagate(ex);
-
+      String imageId = image.getId();
+      Hardware hardwareModel = generateHardwareRequest.isFlavor(hardware.getId());
+      boolean isBaremetal = hardware.getName() == null ? false : hardware.getName().contains(baremetalModelsKey);
+      ServerAppliance workingImage = null;
+
+      //choose the correct image based on the server type baremetal or cloud
```

Comment:
sorry can't understand this.

Can you elaborate a bit more the rationale of this `findWorkingImage` method?

Reply:
Some of the images are of Type ISO and type Image, the ones that are of type ISO are basically an installation of the Linux system and not needed here, this function makes sure to choose the correct boot-able image based on the image from the template at hand